crates/polars-arrow/src/bit_util.rs

```diff
@@ -79,3 +80,178 @@ pub fn ceil(value: usize, divisor: usize) -> usize {
         quot
     }
 }
+
+#[inline]
+const fn get_leading_zeroes(chunk: u64) -> u32 {
+    if cfg!(target_endian = "little") {
+        chunk.trailing_zeros()
+    } else {
+        chunk.leading_zeros()
+    }
+}
+
+#[inline]
+const fn get_leading_ones(chunk: u64) -> u32 {
+    if cfg!(target_endian = "little") {
+        chunk.trailing_ones()
+    } else {
+        chunk.leading_ones()
+    }
+}
+
+fn first_set_bit_impl<I>(mut mask_chunks: I) -> usize
```

Comment:
I think this now becomes redundant with: `find_first_true_false_no_null`, or at least part of that implementation can be reused here.